wsclient/src/main/java/org/hpccsystems/ws/client/utils/Connection.java

```diff
@@ -1055,42 +1057,43 @@ public String sendGetRequest(String uri) throws Exception
      * @throws Exception a {@link java.lang.Exception} object.
      * @return a {@link java.lang.String} object.
      */
-    public String sendHTTPRequest(String uri, String method) throws Exception
+    @WithSpan
+    public String sendHTTPRequest(@SpanAttribute String uri, @SpanAttribute String method) throws Exception
     {
         if (method == null || method.isEmpty())
             throw new Exception ("Must provide valid HTTP method");
 
         URL url = new URL (getBaseUrl() + (uri != null && uri.startsWith("/") ? "" : "/") + uri);
 
-        Span sendHTTPReqSpan = GlobalOpenTelemetry.get().getTracer(BaseHPCCWsClient.PROJECT_NAME)
+        /*Span sendHTTPReqSpan = GlobalOpenTelemetry.get().getTracer(BaseHPCCWsClient.PROJECT_NAME)
                 .spanBuilder(method.toUpperCase() + " " + url.toExternalForm())
                 .setAttribute(ServerAttributes.SERVER_ADDRESS, getHost())
                 .setAttribute(ServerAttributes.SERVER_PORT, Long.getLong(getPort()))
                 .setAttribute(HttpAttributes.HTTP_REQUEST_METHOD, method)
                 .setSpanKind(SpanKind.CLIENT)
                 .startSpan();
-
+*/
         HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection(); //throws IOException
 
         Connection.log.info("Sending HTTP " + method + "Request to:" + url.toString());
 
         if (hasCredentials())
         {
             httpURLConnection.setRequestProperty("Authorization", getBasicAuthString());
-            sendHTTPReqSpan.setAttribute("hasCredentials", true);
-        }
-        else
-        {
-            sendHTTPReqSpan.setAttribute("hasCredentials", false);
+            //sendHTTPReqSpan.setAttribute("hasCredentials", true);
```

Comment:
We discussed this briefly, but does the Zero-code stuff set the currentSpan()? If it does could we grab the current span and use it to add additional attributes?